mlflow/models/python_api.py

```diff
@@ -54,8 +60,7 @@ def build_docker(
 
 def predict(
     model_uri: str,
-    # TODO: This is currently subset of PyfuncInput, ideally we should cover all
-    input_data: Union[str, Dict[str, Any], List[Any], "pd.DataFrame", None] = None,  # noqa: F821
+    input_data: Optional["PyFuncInput"] = None,  # noqa: F821
```

Comment:
Can we import `PyFuncInput` using `if TYPE_CHECKING`?

Reply:
Wow didn't know that smart feature! Not sure if it works for mlflow-skinny build but let me try.

Reply:
oops this seems this doesn't work with sphinx autodoc...
```
/home/circleci/project/mlflow/models/__init__.py:docstring of mlflow.models.python_api.predict:: WARNING: py:class reference target not found: PyFuncInput
```